### ISSUE TITLE
Allow infinite timeout to permanently suppress messages (for a session)

### ISSUE THREAD

**Issue by smathot** [dschreij/QNotifications#6]
Right now, the `timeout` keyword in `QNotifications.__init__()` can be `None`, indicating that subsequent identical messages should not be suppressed, or a positive numeric, indicating that identical messages should be suppressed for an X number of seconds.

The problem (or rather inconvenience) is that you want suppress some message indefinitely, at least for the duration of the application session. An example is here: dschreij/QOpenScienceFramework#5

I therefore propose to:

- Add a Silence button to each message, which will cause the message to be suppressed indefinitely. This button could be to the left of the regular close button, with a clear icon.
- Allow `timeout` to be infinite, either the string 'infinite' or (more elegantly) `math.inf`. In this case, the silence button should be hidden, because the regular close button already silences the message.

What do you think?

**Comment by dschreij** [comment]
Yes, good idea. If the suppression only needs to last the session this should easily possible. I think it should is important though that not *all* notifications can be silenced; the programmer needs an option to indicate which messages are too important to silence beforehand. We wouldn't want the user to miss out on party-stoppin' error notifications.
I think both a string 'infinite' or math.inf will do fine, even though the last requires an extra import again of the math module simply for its inf property, but that doesn't matter much.

**Comment by smathot** [comment]
> I think it should is important though that not all notifications can be silenced; the programmer needs an option to indicate which messages are too important to silence beforehand. We wouldn't want the user to miss out on party-stoppin' error notifications.

Good point. Another keyword, 'silencebutton' (`bool`)?

**Comment by dschreij** [comment]
Yes that seems like a good solution. I'll see if I can add this soon. In context of the Github messages, the only culprit is that the content changes each time if the error is caused by different Github API endpoints. So matching the exact content of the message to see if it needs to be suppressed only gets you so far; if another API endpoint causes the error, it will be visible again. We might need to implement something more intelligent, or a regex, if we want the same type of message to be effectively suppressed, regardless of its contents.

**Comment by smathot** [comment]
I think that for now just suppressing specific messages is sufficient. In the case of GitHub, so far I always get the exact same message. And we can always implement a more sophisticated logic later, right?

**Comment by dschreij** [comment]
Yes! Just thinking ahead, that's all. Let's have a simplistic approach for now.
